keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,730 @@
+
+#### BucketAccessRequest
+
+A user namespaced custom resource representing an object store user and an access policy defining the user’s relation to that storage.  A user creates a `BucketAccessRequest` (BAR) in the app's namespace (which is the same namespace as the `BucketRequest`) A 'BucketAccessRequest' specifies *either* a ServiceAccount or a secret name which contains access policy.  Specifying a ServiceAccount enables provisioners to support cloud provider identity integration with their respective Kubernetes offerings.
+
+```yaml
+apiVersion: cosi.io/v1alpha1
+kind: BucketAccessRequest
+metadata:
+  name:
+  namespace:
+  labels:
+    cosi.io/provisioner: [1]
+  finalizers:
+  - cosi.io/finalizer [2]
+spec:
+  serviceAccountName: [3]
+  bucketRequestName: [4]
+  bucketAccessClassName: [5]
+  bucketAccessName: [6]
+status:
+  accessGranted: [7]
+```
+
+1. `labels`: added by COSI.  Key’s value should be the provisioner name. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
+1. `finalizers`: added by COSI to defer `BucketAccessRequest` deletion until backend deletion ops succeed.
+1. `serviceAccountName`: (optional) the name of a Kubernetes ServiceAccount in the same namespace.  This field is included to support cloud provider identity integration. When supplied, COSI will call the driver to mint new credentials for this ServiceAccount. 
+1. `bucketRequestName`: (required) the name of the `BucketRequest` associated with this access request. From the bucket request, COSI knows the `Bucket` instance and thus the backend bucket and its properties.
+1. `bucketAccessClassName`: (required) name of the `BucketAccessClass` specifying the desired set of policy actions to be set for a user identity or ServiceAccount.
+1. `bucketAccessName`: name of the bound cluster-scoped `BucketAccess` instance. Set by COSI.
+1. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
+
+#### BucketAccess
+
+A cluster-scoped administrative custom resource which encapsulates fields from the `BucketAccessRequest` (BAR) and the `BucketAccessClass` (BAC).  The purpose of the `BucketAccess` (BA) is to serve as an access communication path between provisioners and the central COSI controller.  The COSI controller creates a `BucketAccess` instance for a new `BucketAccessRequest`, and there is a 1:1 mapping between `BucketAccess` and `BucketAccessRequest`.
+
+```yaml
+apiVersion: cosi.io/v1alpha1
+kind: BucketAccess
+metadata: 
+  name: [1]
+  labels:
+    cosi.io/provisioner: [2]
+  finalizers:
+  - cosi.io/finalizer [3]
+ spec:
+  provisioner: [4]
+  bucketInstanceName: [5]
```

Comment:
I forgot where we landed on this discussion, but I remember talking about if this was necessary since BucketAccessRequest contains an implicit reference to Bucket via BucketRequest?

Reply:
Yes, we've gone back and forth on references to other resources within the API. 
We are trying to keep the provisioner sidecar and driver minimally privileged and thus don't want a design that requires the sidecar to retrieve resources in the user's ns. So even though you can get the B via BA.BAR.BR.bucketInstance, we don't want the sidecar to need to GET the BAR or BR.

Reply:
Ack